src/psyclone/tests/psyir/symbols/datasymbol_test.py

```diff
@@ -170,47 +181,51 @@ def test_datasymbol_can_be_printed():
     assert ("s3: DataSymbol<Scalar<REAL, SINGLE>, Import(container='my_mod')>"
             in str(sym3))
 
-    sym3 = DataSymbol("s3", INTEGER_SINGLE_TYPE, constant_value=12)
+    sym3 = DataSymbol("s3", INTEGER_SINGLE_TYPE, initial_value=12)
     assert ("s3: DataSymbol<Scalar<INTEGER, SINGLE>, Automatic, "
-            "constant_value=Literal"
+            "initial_value=Literal"
             "[value:'12', Scalar<INTEGER, SINGLE>]>" in str(sym3))
```

Comment:
I don't think initial_valued symbols can have an AutomaticInterface? Should we prevent them?

Reply:
That's certainly true in Fortran. I guess we choose what we do in PSyIR so we could choose to forbid them?

Reply:
Let's leave it for now. The backend check that you added already catches any problem and it is clear what it is.